recipe/2024-06-private-bicep-registries.md

```diff
@@ -0,0 +1,384 @@
++  authentication?: Record<RegistrySecretConfig>;
+}
+```
+`RegistrySecretConfig` contains registry credential information, we also want users to specify the type of credentials, e.g. username-password, federated credentials.
+
+Option 1: 
+
+Adding a `type` property to `RegistrySecretConfig`
+environment.tsp
+```
+@doc(RegistrySecretType represents the authentication type used to authenticate private Bicep registries.)
+enum RegistryAuthType {
+  @doc("basicAuthentication type is used to represent username and password based authentication and the secretstore resource is expected to have the keys 'username' and 'password'")
+  basicAuthentication
+
+  @doc("azureFederatedIdentity type is used to represent registry authentication using azure federated identity and the secretstore resource is expected to have the keys 'clientId' and 'tenantId'")
+  azureFederatedIdentity
+
+  @doc("awsIRSA type is used to represent registry authentication using AWS IRSA.")
+  awsIRSA
+}
+
+@doc("Registry Secret Configuration used to authenticate to private bicep registries.")
+model RegistrySecretConfig {
+  @doc("The ID of an Applications.Core/SecretStore resource containing credential information used to authenticate private container registry.The keys in the secretstore depends on the type.")
+  secret?: string;
+
+  @doc(RegistryAuthType represents the authentication type used to authenticate private Bicep registries.)
+  type: RegistryAuthType
+}
+```
+Option 2:
+
+Adding support to the `Applications.Core/secretStores` resource to include additional types. Today `secretStores` resource only supports types `certificate` and `generic`.
+
+secretstores.tsp
+
+```diff
+@doc("The type of SecretStore data")
+enum SecretStoreDataType {
+  @doc("Generic secret data type")
+  generic,
+
+  @doc("Certificate secret data type")
+  certificate,
+
++  @doc("basicAuthentication type is used to represent username and password based authentication and the secretstore resource is expected to have the keys 'username' and 'password'")
++  basicAuthentication
++
++  @doc("azureFederatedIdentity type is used to represent registry authentication using azure federated identity and the secretstore resource is expected to have the keys 'clientId' and 'tenantId'")
++  azureFederatedIdentity
++
++  @doc("awsIRSA type is used to represent registry authentication using AWS IRSA.")
++  awsIRSA
+}
+
+```
+I propose adding `type` property to the `RegistrySecretConfig` i.e Option 1, as with the current implementation of the secrets flow between engine-driver,  Option 2 lead to either having driver specific code in engine/secretLoader or driver an making api call to corerp to get secrets information. 
+
+***Bicep Example***
+
+environment.bicep
+```
+"recipeConfig": {
+  "terraform": {
+    ...
+  },
+  "bicep": {
+    "authentication":{
```

Comment:
should these actually be classified as secrets or can they be a little less strict to allow for better troubleshooting for the user?